### PR TITLE
Specialize not-hints

### PR DESCRIPTION
**Pull request by jvoigtlaender**
I'm fairly confident about the hints in the first commit.

The hints in the "Hoist not" commit may be a bit more questionable.

### CODE REVIEW

File: data/hlint.yaml [hunk 1]
```diff
@@ -222,7 +224,9 @@
     - warn: {lhs: any id, rhs: or}
     - warn: {lhs: all id, rhs: and}
     - warn: {lhs: any (not . f) x, rhs: not (all f x), name: Hoist not}
+    - warn: {lhs: any (notElem y) x, rhs: not (all (elem y) x), name: Hoist not}
```
Comment (ndmitchell) [inline data/hlint.yaml:227]: I'd say these are more confusing in the replacement, and no more efficient, so better to leave them as is.

**Review by ndmitchell — COMMENTED** [review]
Agreed that the hoist not ones don't seem to be better, and potentially worse (although not sure sufficiently worse that I'd encourage the other direction). The rest look good though.

**Comment by jvoigtlaender** [comment]
I've just removed the "Hoist not" commit (and force-pushed).

**Comment by ndmitchell** [comment]
Thanks!

**Pull request merged**